source/patterns/@aws-solutions-constructs/aws-kinesisstreams-gluejob/lib/index.ts

```diff
@@ -0,0 +1,186 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { CfnDatabase, CfnJob, CfnJobProps, CfnTable } from '@aws-cdk/aws-glue';
+import { CfnPolicy, Effect, IRole, Policy, PolicyStatement } from '@aws-cdk/aws-iam';
+import { Stream, StreamProps } from '@aws-cdk/aws-kinesis';
+import { Bucket } from '@aws-cdk/aws-s3';
+import { Aws, Construct } from '@aws-cdk/core';
+import * as defaults from '@aws-solutions-constructs/core';
+
+export interface KinesisStreamGlueJobProps {
+  /**
+   * Existing instance of Kineses Data Stream. If not set, it will create an instance
+   */
+  readonly existingStreamObj?: Stream;
+  /**
+   * User provided props to override the default props for the Kinesis Stream.
+   *
+   * @default - Default props are used
+   */
+  readonly kinesisStreamProps?: StreamProps | any;
+  /**
+   * @default
+   * User provides props to override the default props for Glue ETL Jobs. This parameter will be ignored if the
+   * existingGlueJob parameter is set
+   *
+   * This parameter is defined as `any` to not enforce passing the Glue Job role which is a mandatory parameter
+   * for CfnJobProps. If a role is not passed, the construct creates one for you and attaches the appropriate
+   * role policies
+   */
+  readonly glueJobProps?: CfnJobProps | any;
+  /**
+   * Existing GlueJob configuration. If this property is provided, any configuration provided through @glueJobProps
+   */
+  readonly existingGlueJob?: CfnJob;
+  /**
+   * @default
+   * A JSON document defining the schema structure of the records in the data stream. An example of such a
+   * definition as below. Either @table or @fieldSchema is mandatory. If @table is provided then @fieldSchema is ignored
+   * 	"FieldSchema": [{
+   *  	"name": "id",
+   *  	"type": "int",
+   *    "comment": "Identifier for the record"
+   *  }, {
+   *    "name": "name",
+   *    "type": "string",
+   *    "comment": "The name of the record"
+   *  }, {
+   *    "name": "type",
+   *    "type": "string",
+   *    "comment": "The type of the record"
+   *  }, {
+   *    "name": "numericvalue",
+   *    "type": "int",
+   *    "comment": "Some value associated with the record"
+   *  },
+   */
+  readonly fieldSchema?: CfnTable.ColumnProperty [];
+  /**
+   * Glue Database for this construct. If not provided the construct will create a new Glue Database.
+   * The database is where the schema for the data in Kinesis Data Streams is stored
+   */
+  readonly database?: CfnDatabase;
```

Comment:
Can it support L2 `Database` instead of L1 `CfnDatabase` ?

Reply:
L2 construct does not have all the properties of L1. Secondly L2 makes it mandatory to define a database name, while L1 generates a database name.